jmxtrans-output/jmxtrans-output-kafka/src/main/java/com/googlecode/jmxtrans/model/output/kafka/KafkaWriter2.java

```diff
@@ -0,0 +1,123 @@
+package com.googlecode.jmxtrans.model.output.kafka;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.collect.ImmutableMap;
+import com.googlecode.jmxtrans.exceptions.LifecycleException;
+import com.googlecode.jmxtrans.model.OutputWriterAdapter;
+import com.googlecode.jmxtrans.model.Query;
+import com.googlecode.jmxtrans.model.Result;
+import com.googlecode.jmxtrans.model.Server;
+import lombok.Getter;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.Map;
+
+import static com.fasterxml.jackson.databind.annotation.JsonSerialize.Inclusion.NON_NULL;
+
+public class KafkaWriter2 extends OutputWriterAdapter {
+    @Nonnull
+    @Getter
+    private final ObjectMapper objectMapper;
+    @Nonnull
+    @Getter
+    private final Map<String, Object> producerConfig;
+    @Nonnull
+    @Getter
+    private final String topic;
+    private Producer<String, String> producer;
+
+    public KafkaWriter2(@Nonnull ObjectMapper objectMapper, @Nonnull Map<String, Object> producerConfig, @Nonnull String topic) {
+        this.objectMapper = objectMapper;
+        this.producerConfig = producerConfig;
+        this.topic = topic;
+    }
+
+    @Override
+    public void start() throws LifecycleException {
+        producer = new KafkaProducer<String, String>(producerConfig);
```

Comment:
I don't how known how the lifecycle of writers is done, but once closed, the Kafka producer ca not be restarted. Exception that I can move it in the writer constructor.

However, moving the creation of the KafkaProducer outside of the writer (in the writer factory) means exposing (opening) the internals of the writer to the outside. It doesn't look clean to me.

Yet I admit I do not understand the intent behind splitting writer/writer factory.